rest_framework/schemas/inspectors.py

```diff
@@ -509,3 +506,297 @@ def __get__(self, instance, owner):
         inspector = inspector_class()
         inspector.view = instance
         return inspector
+
+
+class OpenAPIAutoSchema(ViewInspector):
+
+    content_types = ['application/json']
+    method_mapping = {
+        'get': 'Retrieve',
+        'post': 'Create',
+        'put': 'Update',
+        'patch': 'PartialUpdate',
+        'delete': 'Destroy',
+    }
+
+    def get_operation(self, path, method):
+        operation = {}
+
+        operation['operationId'] = self._get_operation_id(path, method)
+
+        parameters = []
+        parameters += self._get_path_parameters(path, method)
+        parameters += self._get_pagination_parameters(path, method)
+        parameters += self._get_filter_parameters(path, method)
+        operation['parameters'] = parameters
+
+        request_body = self._get_request_body(path, method)
+        if request_body:
+            operation['requestBody'] = request_body
+        operation['responses'] = self._get_responses(path, method)
+
+        return operation
+
+    def _get_operation_id(self, path, method):
+        """
+        Compute an operation ID from the model, serializer or view name.
+        """
+        # TODO: Allow an attribute/method on the view to change that ID?
```

Comment:
So, ideally, folks would subclass the inspector and assign it as `schema` on the view to customise behavior.

Reply:
That seems a bit overkill just for overriding a single property; why not an `APIView.Meta.openapi_operation_id` attribute, or something renderer-agnostic: `Serializer.Meta.ref_name` [just like in drf-yasg](https://drf-yasg.readthedocs.io/en/stable/custom_spec.html#serializer-meta-nested-class)?

Reply:
Because we want to keep the _schema_ API out of `APIView`, which already has enough to do. 

Now, yes, subclassing **seems** a bit heavy-weight. **But** once the patterns become clear we can adjust the `AutoSchema` inspector class to take the most relevant `kwargs`. So we end up with something that looks like: 

```
class MyView(APIView):
    schema = # Something nice and declarative here...
```

Now, we could inject all sorts of attributes into APIView and Serializer but if we do that then we're tied. There's docs and deprecations to be maintained all across DRF's key modules. 

The great pay-off of insisting that we keep it all inside `schemas` is that we're able to rework the schema handling, as we're doing now, without making any changes elsewhere. (Back when we were first allowing per-view customisation of the schema generation for 3.7, there were lots of "let's just add an attribute here". tl;dr: No. Let's keep it in `schemas`.)